Paintroid/src/main/java/org/catrobat/paintroid/ui/zoomwindow/DefaultZoomWindowController.kt

```diff
@@ -0,0 +1,244 @@
+    }
+
+    override fun onMove(coordinates: PointF) {
+        if (shouldBeInTheRight(coordinates = coordinates)) {
+            setLayoutAlignment(right = true)
+        } else {
+            setLayoutAlignment(right = false)
+        }
+        if (isPointOnCanvas(coordinates.x, coordinates.y)) {
+            if (zoomWindow.visibility == View.GONE) {
+                zoomWindow.visibility = View.VISIBLE
+            }
+            this.coordinates = coordinates
+        } else {
+            dismiss()
+        }
+    }
+
+    override fun getBitmap(bitmap: Bitmap?) {
+        zoomWindowImage.setImageBitmap(coordinates?.let { cropBitmap(bitmap, it) })
+    }
+
+    private fun isPointOnCanvas(pointX: Float, pointY: Float): Boolean =
+        pointX > 0 && pointX < layerModel.width && pointY > 0 && pointY < layerModel.height
+
+    private fun shouldBeInTheRight(coordinates: PointF): Boolean {
+        if (coordinates.x < layerModel.width / 2 && coordinates.y < layerModel.height / 2) {
+            return true
+        }
+        return false
+    }
+
+    private fun setLayoutAlignment(right: Boolean) {
+        val params: RelativeLayout.LayoutParams =
+            zoomWindow.layoutParams as RelativeLayout.LayoutParams
+        if (right) {
+            params.addRule(RelativeLayout.ALIGN_PARENT_RIGHT)
+            params.removeRule(RelativeLayout.ALIGN_PARENT_LEFT)
+        } else {
+            params.addRule(RelativeLayout.ALIGN_PARENT_LEFT)
+            params.removeRule(RelativeLayout.ALIGN_PARENT_RIGHT)
+        }
+        zoomWindow.layoutParams = params
+    }
+
+    private fun cropBitmap(bitmap: Bitmap?, coordinates: PointF): Bitmap? {
+
+        val bitmapWithBackground: Bitmap? = mergeBackground(bitmap)
+
+        // StartX and StartY coordinates
+        // StartX and StartY - windowWidth / 2, without the grey background
+        // But since we are adding the grey background, windowWidth / 2 should be added
+        // So they get cancelled
+        val startX: Int = coordinates.x.roundToInt()
+        val startY: Int = coordinates.y.roundToInt()
+
+        val croppedBitmap: Bitmap? =
+            Bitmap.createBitmap(windowWidth, windowHeight, Bitmap.Config.ARGB_8888)
+
+        val canvas: Canvas? = croppedBitmap?.let { Canvas(it) }
+
+        val paint = Paint()
+        paint.isAntiAlias = true
+
+        val rect = Rect(0, 0, windowWidth, windowHeight)
+        val rectF = RectF(rect)
+
+        canvas?.drawOval(rectF, paint)
+
+        paint.xfermode = PorterDuffXfermode(PorterDuff.Mode.SRC_IN)
+
+        bitmapWithBackground?.let {
+            canvas?.drawBitmap(it,
+                Rect(startX, startY, startX + windowWidth, startY + windowHeight),
+                rect,
+                paint
+            ) }
+
+        return croppedBitmap
+    }
+
+    private fun mergeBackground(bitmap: Bitmap?): Bitmap? {
+
+        // Adding the extra width and height for the grey background
+        val bmOverlay =
```

Comment:
please write out the variable names: bmOverlay => bitmapOverlay. 
This is for better readability of the code a few months down when somebody else needs to take a look at it